src/geo/map/layer-model-base.js

```diff
@@ -55,6 +55,12 @@ var MapLayer = Model.extend({
 
   setError: function (error) {
     this.set('error', error);
+  },
+
+  /**
+   * @Abstract method only needed in Carto and Torque Layers
+   */
+  getSourceId: function () {
```

Comment:
I wouldn't personally add this here given that it's only for CartoDB and Torque layers. Perhaps those two should inherit from a new base class. Something like: CartoDBLayerModel -> DataModel -> LayerModelBase. What do you think?

Reply:
I'm not a fan of inheritance and I think creating intermediate classes complicates the design a lot.

As pointed [here](https://github.com/CartoDB/cartodb.js/pull/1760/files#diff-f99ed960498d5ea8717b3e74d54bfb01R295) I see 2 options.

* Make the method "abstract" so all `layer.getSourceId()` wont fail: This is useful in case we got several loops assuming all the layers have a .getSourceId.

* Just check if the method exists: `layer.getSourceId && layer.getSourceId()`: This is useful if only used in a few places since we don't modify the class inheritance structure.

Reply:
> Make the method "abstract" so all layer.getSourceId() wont fail: This is useful in case we got several loops assuming all the layers have a .getSourceId.

I'm not aware of any loop where we iterate over all layers and do something common for all of them, that's why I think we don't need this method.

> Just check if the method exists: layer.getSourceId && layer.getSourceId(): This is useful if only used in a few places since we don't modify the class inheritance structure.

Again, no need for that so far.

We can always add it to the base class in we case we need it... right?